components/PortfolioModule/finantials.js

```diff
@@ -70,25 +68,34 @@ const getManagerDetailsFromAsset = (asset) => {
 }
 
 export const getPortfolioAssetDetails = (assets, cb) => {
-  let totalAssetRevenue = 0;
-  let totalAssetValue = 0;
-  let totalManagementProfit = 0;
-  const toReturn = assets.map(asset => {
-    totalAssetValue = calculateAssetValue(totalAssetValue, asset);
-    totalAssetRevenue = calculateAssetRevenue(totalAssetRevenue, asset);
-    const managerDetails = getManagerDetailsFromAsset(asset);
-    totalManagementProfit += asset.isAssetManager ? managerDetails.totalProfitAssetManager : 0;
+  const toCalculate = {
+    unrealisedProfitInvestor: 0,
+    unrealisedProfitManager: 0,
+    realisedProfitInvestor: 0,
+    realisedProfitManager: 0,
+    totalAssetValue: 0,
+    totalAssetRevenue: 0,
+  };
+  const assetData = assets.map(asset => {
+    calculateAssetValue(asset, toCalculate);
```

Comment:
these ones do not return any value?

Reply:
Both `calculateAssetValue()` and `calculateAssetRevenue()` make changes to the object declared initially, which contains all the stats, initially all 0. So they both take the current asset in the loop, the object, calculate and update the object.

Reply:
all of these stats

  unrealisedProfitInvestor: 0,
  unrealisedProfitManager: 0,
  realisedProfitInvestor: 0,
  realisedProfitManager: 0,
  totalAssetValue: 0,
  totalAssetRevenue: 0,